sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -267,6 +267,12 @@ private int CompareHeaderDictionaries(SortedDictionary<string, string[]> headers
                     {
                         difference++;
                         descriptionBuilder?.AppendLine($"    <{headerName}> values differ, request <{JoinHeaderValues(requestHeaderValues)}>, record <{JoinHeaderValues(entryHeaderValues)}>");
+
```

Comment:
@pakrym, I found this useful while running tests repeatedly.
Please see if it looks good to you.

Reply:
Consider turning _compareBodies to true for your entire test project or class instead. This might confuse people who explicitly turned the feature off.

It also mutates the field so requests running after the one that failed can start intermittently failing because bodies are compared.

Reply:
Yes, that is one option I considered. However, that seemed like a heavyweight option to _always_ compare bodies.
I'll undo this for now.

Reply:
Why don't we just add a `bool force = false` parameter to `CompareBodies` and [change the condition](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs#L123) to `if (!compareBodies && !force) { return; }`?  Then we can get the more helpful exception without changing any test behavior.

Reply:
>However, that seemed like a heavyweight option to always compare bodies

That's the default behavior and what most other libraries do.

>Then we can get the more helpful exception without changing any test behavior.

I think that the error message that makes it look like we care about bodies is helpful. But if we want to do something like that  we can condition it on `descriptionBuilder != null` and always produce a hint when formatting messages.

Reply:
Using the default behavior seems like a reasonable thing to do.
I see though that [when this behavior was introduced in Core](https://github.com/Azure/azure-sdk-for-net/commit/e87a28d2454ed0653dff60774bf8d38eceeb0ca2#diff-3e56fe99c0dcbd40faae94ab07accc33099823d8da1853ea21b5214aab7611d8), we turned off body comparison in Search and in a few other SDKs.

From reading comments added in other places, I gather that the matcher in some SDKs need to account for some fields that may differ in otherwise correct request/response entries before the option can be enabled.

In a subsequent commit, I will enable the body comparison option after fixing the Search failures.